### PR TITLE
Remove `disable-editing` attribute from the mention view element

### PR DESCRIPTION
**Pull request by vivinkrishna-ni**
# Pull Request

## 🤨 Rationale

Previously, we used `nimble-rich-text-mention-users-view` for two states one is at the time of triggering the mention (Edit mode) and the other one is after committing the mention (View mode) in the editor. However, due to https://github.com/ni/nimble/issues/1716, we wanted to remove the decoration tag (which is configured in Tiptap to render the same element in edit mode) here: https://github.com/ni/nimble/pull/1721. Also, during editing for all mentions we can have the same styling so we decided not to use this element for both edit and view mode. Only after committing in the editor and for viewing in the viewer, this view element can be used. During editing mention node, the styles can be explicitly mentioned in the editor styles.

Relevant discussion: https://github.com/ni/nimble/pull/1721#discussion_r1440837527

Will close the following issues upon merging this PR:
1. fixes https://github.com/ni/nimble/issues/1659
2. fixes https://github.com/ni/nimble/issues/1716

## 👩‍💻 Implementation

- Removed `disable-editing` attribute from the view element base class
- Removed styles related to `disable-editing` and other CSS cleanups in view element
- Added [`decorationClass`](https://tiptap.dev/docs/editor/api/utilities/suggestion#decoration-class) in the editor to style the default suggestion `span` in the editor and added necessary styles to it. 

## 🧪 Testing

- Removed the view element edit mode test case from the Chromatic matrix test
- Manually tested the mention view element in editor and viewer

## ✅ Checklist

- [x] I have updated the project documentation to reflect my changes or determined no changes are needed.

### CODE REVIEW

**Review by m-akinc — APPROVED** [review]

**Review by jattasNI — APPROVED** [review]

File: packages/nimble-components/src/rich-text/editor/styles.ts [hunk 1]
```diff
@@ -80,6 +80,15 @@ export const styles = css`
...
     }
 
+    ${/* Shared styles for all mention views at edit time. */ ''}
+    .nimble-mention-view-edit {
+        color: ${bodyFontColor};
+    }
+
+    :host([disabled]) .nimble-mention-view-edit {
+        color: ${bodyDisabledFontColor};
+    }
```
Comment (rajsite) [inline packages/nimble-components/src/rich-text/editor/styles.ts:90]: These are styles associated with the ProseMirror inserted content so for the styles to be in document order it should be aligneed with those down below:


```suggestion
    .ProseMirror .nimble-mention-view-edit {
        color: ${bodyFontColor};
    }

    :host([disabled]) .ProseMirror .nimble-mention-view-edit {
        color: ${bodyDisabledFontColor};
    }
```

File: packages/nimble-components/src/rich-text/editor/styles.ts [hunk 2]
```diff
@@ -80,6 +80,15 @@ export const styles = css`
         border: ${borderWidth} solid rgba(${borderRgbPartialColor}, 0.1);
     }
 
+    ${/* Shared styles for all mention views at edit time. */ ''}
```
Comment (rajsite) [inline packages/nimble-components/src/rich-text/editor/styles.ts:83]: There is one selector in the ProseMirror inserted content section below that looks liek:

```
li > p {
        margin-block: 0;
    }
```

And should be updated to:
```
.ProseMirror li > p {
        margin-block: 0;
    }
```

**Review by rajsite — COMMENTED** [review]
Looks good just some minor style tweaks

File: packages/nimble-components/src/rich-text/editor/styles.ts (hunk 1 above)
Comment (vivinkrishna-ni) [reply packages/nimble-components/src/rich-text/editor/styles.ts]: Moved the styles below to anchor styles and updated as suggested.

File: packages/nimble-components/src/rich-text/editor/styles.ts (hunk 2 above)
Comment (vivinkrishna-ni) [reply packages/nimble-components/src/rich-text/editor/styles.ts]: Thanks for pointing it out. I have updated the same.

**Review by rajsite — APPROVED** [review]

**Pull request merged**